In [1]:
# load dataset filenames from data\train_filenames.lst and data\test_filenames.lst
train_filenames_path = "data/train_filenames.lst"
with open(train_filenames_path, "r") as f:
    train_filenames = f.read().splitlines()
test_filenames_path = "data/test_filenames.lst"
with open(test_filenames_path, "r") as f:
    test_filenames = f.read().splitlines()

print(f"Number of train images: {len(train_filenames)}")
print(f"Number of test images: {len(test_filenames)}")
print(f"Total number of images: {len(train_filenames) + len(test_filenames)}")


Number of train images: 45337
Number of test images: 5044
Total number of images: 50381


In [16]:
# Unzip the dataset if not already unzipped (data/images_200m.zip)
import os
import zipfile
import numpy as np

# check if data/images_200m.csv exists, if not unzip data/images_200m.zip
if not os.path.exists("data/images_200m.csv"):
    print("Unzipping dataset...")
    dataset_zip_path = "data/images_200m.zip"
    with zipfile.ZipFile(dataset_zip_path, "r") as zip_ref:
        zip_ref.extractall("data")
        print("Dataset is ready.")
else:
    print("Dataset is already unzipped")

# Load the dataset as np array

images_200m = np.loadtxt("data/images_200m.csv", delimiter=",", dtype=np.uint16)
print(f"Dataset shape: {images_200m.shape}")




Dataset is already unzipped


C:\Users\uceda\AppData\Local\Temp\ipykernel_16828\2981227279.py:18: DeprecationWarning: loadtxt(): Parsing an integer via a float is deprecated.  To avoid this warning, you can:
    * make sure the original data is stored as integers.
    * use the `converters=` keyword argument.  If you only use
      NumPy 1.23 or later, `converters=float` will normally work.
    * Use `np.loadtxt(...).astype(np.int64)` parsing the file as
      floating point and then convert it.  (On all NumPy versions.)
  (Deprecated NumPy 1.23)
  images_200m = np.loadtxt("data/images_200m.csv", delimiter=",", dtype=np.uint16)


Dataset shape: (45337, 5292)


In [17]:
# Check the first 5 rows of the dataset
print("First 5 rows of the dataset:")
print(images_200m[:5])

First 5 rows of the dataset:
[[ 64  91  60 ...   0   0   0]
 [315 316 262 ...   0   0   0]
 [413 450 432 ...   0   0   0]
 [654 613 498 ...   0   0   0]
 [123 139 113 ...   0   0   0]]


Now `images_200m` contains the flattened version of all images.

    Shape originally: (N, 12, 21, 21) → flattened version: (N, 12×21×21) = (N, 5292)

The flattening order follows NumPy's default row-major (C-order), meaning the values are ordered as:
    
    band_0_row_0_col_0, band_0_row_0_col_1, ..., band_0_row_20_col_20,
    band_1_row_0_col_0, ..., band_1_row_20_col_20,
    ...
    band_11_row_0_col_0, ..., band_11_row_20_col_20



In [2]:
species_list = [
    'Abies_alba', 
    'Acer_pseudoplatanus',
    'Alnus_spec', 
    'Betula_spec', 
    'Cleared', 
    'Fagus_sylvatica', 
    'Fraxinus_excelsior', 
    'Larix_decidua', 
    'Larix_kaempferi', 
    'Picea_abies', 
    'Pinus_nigra', 
    'Pinus_strobus', 
    'Pinus_sylvestris', 
    'Populus_spec', 
    'Prunus_spec', 
    'Pseudotsuga_menziesii', 
    'Quercus_petraea', 
    'Quercus_robur', 
    'Quercus_rubra', 
    'Tilia_spec'
]

In [3]:
# load labels from data/labels_one_hot.csv as np array
import pandas as pd
# with 'one_hot' as header
loaded_labels = pd.read_csv('data/labels_one_hot.csv', header=0)
print(loaded_labels.shape)
print(loaded_labels.dtypes)  
print(loaded_labels.head())

(45337, 1)
one_hot    str
dtype: object
                                             one_hot
0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...
1  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...
2  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...
3  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...
4  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...


In [9]:
import ast

# Parse the string lists into actual lists, then expand into columns
labels_expanded = loaded_labels['one_hot'].apply(ast.literal_eval)
one_hot_df = pd.DataFrame(labels_expanded.tolist(), columns=species_list)

one_hot_df.head()

,Abies_alba,Acer_pseudoplatanus,Alnus_spec,Betula_spec,Cleared,Fagus_sylvatica,Fraxinus_excelsior,Larix_decidua,Larix_kaempferi,Picea_abies,Pinus_nigra,Pinus_strobus,Pinus_sylvestris,Populus_spec,Prunus_spec,Pseudotsuga_menziesii,Quercus_petraea,Quercus_robur,Quercus_rubra,Tilia_spec
0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
3,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
4,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


In [7]:
loaded_labels.head()

,one_hot
0,"[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..."
1,"[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..."
2,"[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..."
3,"[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..."
4,"[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..."
